src/java.desktop/share/classes/sun/swing/CachedPainter.java

```diff
@@ -314,8 +314,9 @@ public int getHeight(ImageObserver observer) {
 
         @Override
         public Image getResolutionVariant(double destWidth, double destHeight) {
-            int w = (int) Math.ceil(destWidth);
-            int h = (int) Math.ceil(destHeight);
+            int w = (int) Math.floor(destWidth + 0.5);
+            int h = (int) Math.floor(destHeight + 0.5);
```

Comment:
Does this problem only affect windows? Will this change in shared code affect the painting of other platforms?

Reply:
I think this function was added by the fix for [JDK-8143064](https://bugs.openjdk.org/browse/JDK-8143064) : Icons are not properly rendered with Windows L&F on HiDPI display

which was for Windows.
I tested my change on other platforms as well , didn't see any regression.